flink-python/src/main/java/org/apache/flink/streaming/api/runners/python/beam/BeamPythonFunctionRunner.java

```diff
@@ -250,7 +250,10 @@ public void open(PythonConfig config) throws Exception {
 		Struct pipelineOptions = PipelineOptionsTranslation.toProto(portableOptions);
 
 		if (memoryManager != null && config.isUsingManagedMemory()) {
-			Preconditions.checkArgument(managedMemoryFraction > 0 && managedMemoryFraction <= 1.0);
+			Preconditions.checkArgument(managedMemoryFraction > 0 && managedMemoryFraction <= 1.0,
+				"The configured managed memory fraction for Python worker process must be within (0, 1], was: %s. " +
+				"It maybe because the consumer type \"Python\" was missing or set to 0 for the config option \"taskmanager.memory.managed.consumer-weights\"." +
```

Comment:
PYTHON has a default weight 30 so missing setting it will not result in this problem.

Reply:
typo: maybe -> may be

Reply:
I meant missing Python entry, for example, users configure *taskmanager.memory.managed.consumer-weights* as *DATAPROC:100*

Reply:
Yes you are right. That is also possible to happen and will result in problems.